### ISSUE TITLE
liquid fuel of next stage is returned by stage:liquidfuel

### ISSUE THREAD

**Issue by tfischer4765** [KSP-KOS/KOS#1124]
When I added a booster stage to my rocket, I noticed it broke my staging logic. Adding a bit of debug output revealed that the liquid fuel on the _next_ stage is returned by querying stage:liquidfuel on the current stage. This breaks simple staging logic like this to no end:

    when stage:liquidfuel < 1 and stage:solidfuel < 1 and stage:number >0 then {
        stage.
        preserve.
    }

My understanding would be that on a solid fuel booster stage, stage:liquidfuel should return 0. Am I wrong?

![screenshot1](https://cloud.githubusercontent.com/assets/1536481/9350775/c868f530-4653-11e5-9f06-06d3065a3821.png)

**Comment by TDW89** [comment]
Bring up the stock resource panel and tick stage only. KSP sometimes gets confused when calculating what should be in which stage.
Unless it has changed recently kOS is just returning the value provided by the KSP API.

**Comment by tfischer4765** [comment]
I've since been able to reproduce the problem with liquid fuel boosters as well. Strangely enough, when I tick stage only, the fuel doesn't show up at all any more (blazing engines nonwithstanding). I think it might have something to do with the way the separators are staged, because when I stage the separator separately (hehe), the problem doesn't occur. Curiously enough, separation of stage 1 works  like a charm. Might the craft file be corrupt?

**Comment by tfischer4765** [comment]
Nope, not a corrupt craft file. I built a similar craft and got the same results except this time the liquid fuel shows on the stage only view. I fear this will be something Squad has to fix for us.

**Comment by TDW89** [comment]
the code:
```
for part in ship:parts {
 print part:name + ":  " + part:stage.
}
```
might help you find where it's getting mixed up

if you are after staging logic you can use `set old_maxthrust to 10^10. //A big number` at the start of your code then in the main loop
```
if ship:maxthrust > old_maxthrust {
 stage.
wait 0.1.
 set old_maxthrust to ship:maxthrust.
}.
```
That way it doesn't matter what fuel you use it just stages when the trust drops (from an engine flame out)

**Comment by tfischer4765** [comment]
are you quite sure about ship:maxthrust > old_maxthrust ? I might overlook something but that condition doesn't look ilke it triggers when the maxthrust _drops_ but rather when it goes up.

**Comment by TDW89** [comment]
um... oops that should be a less than `<`

**Comment by hvacengi** [comment]
This is already updated in the develop branch, there just hasn't been an update released since I pushed it at the end of June.  Watch for the fix in the next release.

Ironically, in testing I realized on other issue.  Stock KSP has a bug that allows liquid fuel to flow across radial decouplers if any part other than a fuel tank is mounted to the decoupler.  

So if you put a SRB on radially, kOS will report that liquid fuel as available to the stage.  And in fact it should as you can then radially mount an engine to the SRB, and the liquid engine will burn the fuel from the tank the decoupler is mounted to.  The "stock bux fix" mod by ClawKSP fixes this issue however.

**Comment by hvacengi** [comment]
I'm going to close this thread because it appears to have been fixed in PR #1082.
